### PR TITLE
Paper Vote functionality

### PR DESCRIPTION
**Pull request by sobchenko**

### CODE REVIEW

File: src/AdminBundle/Controller/PaperVoteController.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     * @param Request $request
+     *
+     * @return array | bool | RedirectResponse
+     */
+    public function newAction(User $user, VoteSettings $voteSettings, Request $request)
+    {
+        //TODO: this action needs refactoring
+        $balanceVotes = $voteSettings->getVoteLimits() -
+            $this->getDoctrine()->getRepository(User::class)
+                ->getUserVotesBySettingVote($voteSettings, $user);
```
Comment (smagolexandr) [inline src/AdminBundle/Controller/PaperVoteController.php:45]: make it in 1 line. its no way to refactor this

File: src/AdminBundle/Controller/PaperVoteController.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        }
+
+        $projects = $this->getDoctrine()->getManager()->getRepository(Project::class)->createQueryBuilder('p')
+            ->andWhere('p.voteSetting = :vs')
+            ->setParameter('vs', $voteSettings)
+            ->getQuery()
+            ->getResult();
+
+        $addForm = $this->createFormBuilder()
+            ->add('blankNumber', null, ['label' => 'Номер бланку'])
```
Comment (smagolexandr) [inline src/AdminBundle/Controller/PaperVoteController.php:58]: add asserts maybe

File: src/AdminBundle/Controller/UserController.php [hunk 3]
```diff
@@ -106,47 +168,49 @@ public function newAction()
...
             return $this->redirectToRoute('admin_users');
         }
 
-        $deleteForm = $this->createDeleteForm($id);
+        $deleteForm = $this->createDeleteForm($user->getId());
 
-        $em    = $this->get('doctrine.orm.entity_manager');
-        $dql   = "SELECT a FROM AppBundle:Project a LEFT JOIN a.voteSetting vs LEFT JOIN vs.location l WHERE l.city = :city ORDER BY a.id DESC";
+        /** @var VoteSettings[] $voteSettings */
+        $voteSettings = $this->getDoctrine()->getRepository('AppBundle:VoteSettings')->getVoteSettingByUserCity($user);
```
Comment (smagolexandr) [inline src/AdminBundle/Controller/UserController.php:197]: `'AppBundle:VoteSettings'` to VoteSettings::class

File: src/AdminBundle/Controller/UserController.php [hunk 4]
```diff
@@ -209,44 +274,49 @@ public function editAction($id, Request $request)
...
         $editForm->handleRequest($request);
 
-        if ($editForm->isValid()) {
+        $errors = $this->get('validator')->validate($entityUserModel->getUser(), null, ['admin_user_put']);
+
+        if ($editForm->isValid() && count($errors) === 0) {
             $em->flush();
 
-            return $this->redirect($this->generateUrl('admin_users_edit', array('id' => $id)));
+            return $this->redirect($this->generateUrl('admin_users_edit', array('id' => $user->getId())));
```
Comment (smagolexandr) [inline src/AdminBundle/Controller/UserController.php:303]: array() to []

File: src/AdminBundle/Controller/UserController.php [hunk 5]
```diff
@@ -209,44 +274,49 @@ public function editAction($id, Request $request)
...
             'delete_form' => $deleteForm->createView(),
+            'errors' => $errors
         );
     }
 
     /**
      * @Route("/projects/{id_project}/like/user/{id_user}", name="admin_projects_like", requirements={"id_project" = "\d+", "id_user" = "\d+"})
      * @Template()
+     *
+     * @deprecated
```
Comment (smagolexandr) [inline src/AdminBundle/Controller/UserController.php:318]: why depracated?

File: src/AdminBundle/Controller/UserController.php (hunk 3 above)
Comment (sobchenko) [reply src/AdminBundle/Controller/UserController.php:197]: you wrote this line, and you can change it

File: src/AdminBundle/Controller/UserController.php (hunk 5 above)
Comment (sobchenko) [reply src/AdminBundle/Controller/UserController.php:318]: It's not necessary anymore

File: src/AdminBundle/Controller/PaperVoteController.php (hunk 1 above)
Comment (sobchenko) [reply src/AdminBundle/Controller/PaperVoteController.php:45]: I wrote about refactoring for all action

**Pull request merged**